### DISCUSSION TITLE
Rust for 16bit 8086

### DISCUSSION THREAD

**Discussion by toncho11** [jbruchon/elks#1782] [Ideas]
Main thread:
https://www.reddit.com/r/rust/comments/zzqfn8/how_can_i_compile_rust_for_16bit_x86_intel_8086/

Some of the projects in the above thread:
https://github.com/Serentty/rusty-dos

https://github.com/cyrozap/rv51
You can run a Rust program on the rv51 https://github.com/cyrozap/rv51/tree/master/examples/rust by first compiling it to RISC-V and then executing through the rv51 emulator.

@ghaerr
@tkchia

**Comment by toncho11** [comment]
So if this risc V emulator https://github.com/cyrozap/rv51/blob/master/src/main.S can be re-written to target the 8086 cpu then we will have a Rust to 8086 "compiler".

**Reply by sleirsgoevy** [reply]
I guess WebAssembly/wasm2c make more sense as an intermediate target.

**Reply by toncho11** [reply]
Yes. It might actually work. I should try a hello world rust app to wasm and convert it to C. Then use ELKS 16 gcc compiler to produce an ELKS compatible executable.

**Reply by toncho11** [reply]
So here is a list of rust command line utilities: [https://gist.github.com/sts10/daadbc2f403bdffad1b6d33aff016c0a](https://gist.github.com/sts10/daadbc2f403bdffad1b6d33aff016c0a). Some of the simpler ones might be successfully converted.

**Comment by Oichkatzelesfrettschen** [comment]
Otherwise, it looks like someone will need to solve the llvm issues for llvm to be a viable 8086/8087; 16-bit compiler. 

Get that out of the way and then work on 16-bitifying Rust.
